src/templates/motion-category-template.js

```diff
@@ -0,0 +1,283 @@
+import React from "react"
+import SEO from "../components/seo"
+import Layout from "../components/layout"
+import { motionCategorySlug } from "../utils"
+import MotionStatusChip from "../components/motionStatusChip"
+import Breadcrumb from "../components/motion/breadcrumb"
+import { Link } from "gatsby"
+import HiddenOnMobile from "../components/hiddenOnMobile"
+import { media } from "../styles"
+
+export const query = graphql`
+  query($sub_cat: String!) {
+    allMotionYaml(filter: { sub_cat: { eq: $sub_cat } }) {
+      totalCount
+      edges {
+        node {
+          fields {
+            slug
+          }
+          id
+          registration_no
+          name
+          status
+        }
+      }
+    }
+  }
+`
+
+class MotionCategoryPage extends React.Component {
+  constructor(props) {
+    super(props)
+    this.allMotions = props.data.allMotionYaml.edges.map(e => e.node)
+    this.sortBy = "asc"
+    this.filterBy = "all"
+    this.state = {
+      motions: this.allMotions,
+    }
+  }
+
+  sort = event => {
+    this.sortBy = event.target.value
+    this.updateMotions()
+  }
+
+  filter = event => {
+    this.filterBy = event.target.value
+    this.updateMotions()
+  }
+
+  updateMotions = () => {
+    const motions = [...this.allMotions]
+    if (this.sortBy === "asc") {
+      motions.sort((a, b) => parseInt(a.id) > parseInt(b.id))
+    } else if (this.sortBy === "desc") {
+      motions.sort((a, b) => parseInt(a.id) < parseInt(b.id))
+    } else if (this.sortBy === "regis_no_desc") {
+      motions.sort(this.regisNoDesc)
+    } else if (this.sortBy === "regis_no_asc") {
+      motions.sort(this.regisNoAsc)
+    }
```

Comment:
The branching is a bit lengthly. Maybe, we can create a sorter object containing all the sort type attributes.

For example,
```
const sorter  = {
   "asc": (a, b) => ... 
   "desc": (a, b) => ... 
   "regis_no_desc": (a, b) =>
}

motions.sort(sorter[this.sortBy])
```

Reply:
Definitely better. I was looking at it last night too, but didn't have time to consider it well enough. 😆

Thanks for all suggestions, will update soon.

Reply:
imho, this is a minor improvement. we can also do it later.